string.c

```diff
@@ -3339,7 +3337,9 @@ rb_str_concat_literals(size_t num, const VALUE *strary)
     if (UNLIKELY(num == 1)) return rb_str_resurrect(strary[0]);
 
     for (i = 0; i < num; ++i) { len += RSTRING_LEN(strary[i]); }
-    if (LIKELY(len < MIN_PRE_ALLOC_SIZE)) {
+    // if the new combined string can remain embedded in the first strings slot
+    // use rb_str_resurrect to avoid an allocation.
+    if ((size_t)len <= (rb_gc_obj_slot_size(strary[0]) - sizeof(struct RBasic))) {
         str = rb_str_resurrect(strary[0]);
```

Comment:
`rb_str_resurrect` duplicates the first string anyways (i.e. creates a new string). We have the new length in `len`, why not just directly VWA a new string with capacity of `len` (+ perhaps the terminator)?

Reply:
`rb_str_buf_new` already takes into account the terminator.